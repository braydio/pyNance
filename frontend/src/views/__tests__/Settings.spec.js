// @vitest-environment jsdom
import { describe, it, expect, vi } from 'vitest'
import { shallowMount, flushPromises } from '@vue/test-utils'
import SettingsView from '../Settings.vue'

vi.mock('axios', () => ({
  default: {
    get: vi.fn().mockResolvedValue({ data: { themes: [], current_theme: '' } }),
    post: vi.fn().mockResolvedValue({})
  }
}))

describe('Settings.vue', () => {
  it('matches snapshot', async () => {
    const wrapper = shallowMount(SettingsView, {
      global: {
        stubs: ['BasePageLayout', 'PageHeader', 'SettingsIcon']
      }
    })
    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
