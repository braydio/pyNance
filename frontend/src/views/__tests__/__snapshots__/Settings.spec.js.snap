// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Settings.vue > matches snapshot 1`] = `"<base-page-layout-stub padding="true" gap="gap-6"></base-page-layout-stub>"`;
